sda/cmd/ingest/ingest_test.go

```diff
@@ -36,13 +36,31 @@ func (suite *TestSuite) SetupTest() {
 	pubKeyList = append(pubKeyList, publicKey)
 
 	tempDir := suite.T().TempDir()
-	privateKeyFile, err := os.Create(fmt.Sprintf("%s/c4fg.key", tempDir))
+
+	// Write the first private key file
+	privateKeyFile1, err := os.Create(fmt.Sprintf("%s/c4gh1.key", tempDir))
 	assert.NoError(suite.T(), err)
-	err = keys.WriteCrypt4GHX25519PrivateKey(privateKeyFile, privateKey, []byte("password"))
+	err = keys.WriteCrypt4GHX25519PrivateKey(privateKeyFile1, privateKey, []byte("c4ghpass"))
 	assert.NoError(suite.T(), err)
-	viper.Set("c4gh.filepath", fmt.Sprintf("%s/c4fg.key", tempDir))
-	viper.Set("c4gh.passphrase", "password")
+	privateKeyFile1.Close()
 
+	// Write the second private key file
+	privateKeyFile2, err := os.Create(fmt.Sprintf("%s/c4gh2.key", tempDir))
+	assert.NoError(suite.T(), err)
+	err = keys.WriteCrypt4GHX25519PrivateKey(privateKeyFile2, privateKey, []byte("c4ghpass"))
+	assert.NoError(suite.T(), err)
+	privateKeyFile2.Close()
+
+	viper.Set("c4gh.privateKeys", []map[string]string{
+		{
+			"filePath":   fmt.Sprintf("%s/c4gh1.key", tempDir),
+			"passphrase": "c4ghpass",
+		},
+		{
+			"filePath":   fmt.Sprintf("%s/c4gh2.key", tempDir),
+			"passphrase": "c4ghpass",
+		},
```

Comment:
While I can see what you are trying to do, you are not thinking it all the way through. And that reflects in the changes to the tests below.

What you have here is two files with different names but the contents is identical since only one keypair is generated above.

Everything form line 33 down to here needs to be reworked, use a loop so you can generate multiple c4gh keys without having to duplicate a lot of the code.

Reply:
This part has been refactored substantially, it used a helper function CreatePrivateKeyFile to reduce the repeated code in various testing functions. The reason why creating key files are not added in the SetupTest for `config.go` since different test functions used them differently. But with the addition of the helper function, duplication is mostly avoided. 
See the commit b81ce66c for details

Reply:
That the keys are used differently in the separate tests is not an issue since the keys will be recreated before each test  starts.
But we can leave that for the time when we refactor this test file, because it is needed.